server/src/main/scala/org/dbpedia/extraction/server/Server.scala

```diff
@@ -13,119 +13,196 @@ import org.dbpedia.extraction.util.{ExtractionRecorder, Language}
 import scala.reflect.{ClassTag, classTag}
 
 class Server(
-  private val password : String,
-  languages : Seq[Language],
-  val paths: Paths,
-  mappingTestExtractors: Seq[Class[_ <: Extractor[_]]],
-  customTestExtractors: Map[Language, Seq[Class[_ <: Extractor[_]]]])
-{
-    val managers: SortedMap[Language, MappingStatsManager] = {
-      val tuples = languages.map(lang => lang -> new MappingStatsManager(paths.statsDir, lang))
-      SortedMap(tuples: _*)
+              private val password: String,
+              languages: Seq[Language],
+              val paths: Paths,
+              mappingTestExtractors: Seq[Class[_ <: Extractor[_]]],
+              customTestExtractors: Map[Language, Seq[Class[_ <: Extractor[_]]]]) {
+  val managers: SortedMap[Language, MappingStatsManager] = {
+    val tuples = languages.map(lang => lang -> new MappingStatsManager(paths.statsDir, lang))
+    SortedMap(tuples: _*)
+  }
+
+  val redirects: Map[Language, Redirects] = {
+    managers.map(manager => (manager._1, buildTemplateRedirects(manager._2.wikiStats.redirects, manager._1)))
+  }
+
+  val extractor: ExtractionManager = new DynamicExtractionManager(managers(_).updateStats(_), languages, paths, redirects, mappingTestExtractors, customTestExtractors)
+
+  extractor.updateAll
+
+  def adminRights(pass: String): Boolean = password == pass
+
+  /**
+   * Extract using a specific extractor only
+   * This method creates a temporary extraction manager with only the requested extractor
+   */
+  def extractWithSpecificExtractor(source: Source, destination: Destination, language: Language, extractorName: String): Unit = {
+    val logger = Server.logger
+
+    try {
+      // Get the manager for this language
+      val manager = managers.get(language) match {
+        case Some(mgr) => mgr
+        case None => throw new IllegalArgumentException(s"No manager configured for language ${language.wikiCode}")
+      }
+
+      // Get all available extractors for this language from the custom test extractors
+      val customExtractorsForLang = customTestExtractors.getOrElse(language, Seq.empty)
+      val mappingExtractors = mappingTestExtractors
+
+      // Find the specific extractor class by name
+      val extractorClass = (customExtractorsForLang ++ mappingExtractors).find { extractorClass =>
+        extractorClass.getSimpleName == extractorName ||
+          extractorClass.getSimpleName == (extractorName + "Extractor") ||
+          extractorClass.getName.endsWith("." + extractorName)
+      } match {
+        case Some(clazz) => clazz
+        case None => throw new IllegalArgumentException(s"Extractor '$extractorName' not found for language ${language.wikiCode}")
+      }
+
+      logger.info(s"Creating individual extraction manager for extractor: ${extractorClass.getSimpleName}")
+
+      // Create a temporary extraction manager with only this extractor
+      val singleExtractorManager = new DynamicExtractionManager(
```

Comment:
We could use a loading cache here, e.g. see https://github.com/google/guava/wiki/cachesexplained to optimise memory/cpu. There is no need for the cache to expire, but only a max size of e.g 100 instances